test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -0,0 +1,445 @@
+        );
+
+    });
+
+    describe("Initialization", () => {
+        it("points to the DebtRegistry passed in through the constructor", async () => {
+            await expect(termsContract.debtRegistry.callAsync()).to.eventually.equal(mockRegistry.address);
+        });
+        it("points to the RepaymentRouter passed in through the constructor", async () => {
+          await expect(termsContract.repaymentRouter.callAsync()).to.eventually.equal(router.address);
+        });
+        it("points to the token contract passed in through the constructor", async () => {
+          await expect(termsContract.repaymentToken.callAsync()).to.eventually.equal(mockToken.address);
+        });
+    });
+
+    describe("#registerRepayment", () => {
+
+        describe("agent who is not RepaymentRouter calls registerRepayment", () => {
+
+            it("should throw", async () => {
+              await expect(termsContract.registerRepayment.sendTransactionAsync(
+                ARBITRARY_AGREEMENT_ID,
+                PAYER,
+                BENEFICIARY,
+                Units.ether(1),
+                mockToken.address,
+                { from: ATTACKER } // the message is coming from an attacker.
+              )).to.eventually.be.rejectedWith(REVERT_ERROR);
+            });
+
+        });
+
+        describe("RepaymentRouter calls registerRepayment", () => {
+
+            before(async () => {
+                await mockRegistry.mockGetBeneficiaryReturnValueFor.sendTransactionAsync(
+                    ARBITRARY_AGREEMENT_ID,
+                    BENEFICIARY,
+                );
+
+                await mockRegistry.mockGetTermsReturnValueFor.sendTransactionAsync(
+                    ARBITRARY_AGREEMENT_ID,
+                    termsContract.address,
+                    TERMS_CONTRACT_PARAMETERS,
+                );
+
+                await mockRegistry.mockGetTermsContractReturnValueFor.sendTransactionAsync(
+                    ARBITRARY_AGREEMENT_ID,
+                    termsContract.address,
+                );
+            });
+
+            describe("...with a different `tokenAddress` than expected by the terms contract", () => {
+
+                before(async () => {
+                    await expect(router.repay.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        Units.ether(1),
+                        ATTACKER // this is not the address it's expecting.
+                    )).to.eventually.be.rejectedWith(REVERT_ERROR);
+                });
+
+                it("should not record the repayment", async () => {
+                    await expect(termsContract.getValueRepaidToDate.callAsync(
+                      ARBITRARY_AGREEMENT_ID
+                    )).to.eventually.bignumber.equal(Units.ether(0));
+                });
+            });
+
+            describe("...with terms contract's expected `tokenAddress`", () => {
+
+              const AMOUNT = Units.ether(1);
```

Comment:
Nit: should be aligned with `before` and `it` statements.

Reply:
Good eye! Added a pivotal task for running `prettier` on all files as a default.